### DISCUSSION TITLE
I'm helping!

### DISCUSSION THREAD

**Discussion by waynevanson** [gcanti/fp-ts#1411] [Q&A] [answered]
I'm looking forward to version 3!

Do you have a todo list? Whether it's here or with the other libraries for compatibility.

**Comment by gcanti** [comment, marked as answer]
Thank you @waynevanson,

I have a todo list containing general tasks, like:

# "weird" names

In order to be more friendly I'd like to rename some APIs. 

Examples in v2.10:

- `cons` -> `prepend`
- `snoc` -> `append`
- `getDualSemigroup` -> `reverse`
- `fold` -> `concatAll`

# unsafe signatures

Scan the codebase looking for possibly unsafe signatures.

Example (`ReadonlyRecord`'s singleton)

```ts
declare function singleton<K extends string, A>(k: K, a: A): ReadonlyRecord<K, A>

declare const k: 'a' | 'b'

// const x1: Readonly<Record<"a" | "b", number>>
const x1 = singleton(k, 1)
```

# type parameters order

In v3 I'm following this convention: type parameters should be declared in order of appearance.

(a linting rule would help)

# make `pipe`able all the things

Or document why there's an exception to this rule

**Reply by gkamperis** [reply]
completely personal preference/vote but if I was to choose I would keep the "weird" names of that specific list...

**Reply by dszlachta** [reply]
@gcanti I'm thinking about creating a PR from time to time adding more examples, would that be helpful? I'm frontend developer with practically no Haskell knowledge (yet) and sometimes I have to browse a few different sites to find out how to use some functions.

**Reply by jituanlin** [reply]
What can we do to contribute to 3.0?
